packages/webapp/pages/my-feed.tsx

```diff
@@ -12,7 +14,11 @@ const seo: NextSeoProps = {
   ...defaultSeo,
 };
 
-const MyFeed = (): ReactElement => <NextSeo {...seo} />;
+const MyFeed = (): ReactElement => {
+  useFeature(Features.EngagementLoopJuly2023Upvote);
```

Comment:
Why is this invoked here?
We already invoke it in the hook right?

Reply:
One in the hook is only to check if experiment is on, this is the activation that should happen only on my feed.

Reply:
Hmm but does that not enroll people already?

Reply:
`useFeatureIsOn` does not enroll I think, I'll check

Reply:
@rebelchris hmm looks like it does just call `evalFeature` under the hood, which does enroll https://github.com/growthbook/growthbook/blob/33f5f7c1c7087336cf70513b9ceab68605ad0952/packages/sdk-js/src/GrowthBook.ts#L468-L470

kinda funny they would call hook `useFeatureIsOn` then because I would not expect it to enroll. 

Also since `useFeatureIsOn` enrolls user we can not really use it in `usePostFeedback` hook since it could enroll user in experiment from reading history page which is not what experiment requires.

Reply:
Ah good point! Something to solve indeed

Reply:
I asked the question on growthbook slack.

Reply:
Reply from slack.

I added following as example:
```jsx
const isFeatureEnabled = useIsFeatureEnabled('feature') // just check if feature is enabled

{isFeatureEnabled && <ButtonA />}
{isFeatureEnabled && <ButtonB />}
<ButtonC onClick={() => {
    growthbook.feature('feature') // enable feature and bucket user
} />
```

> To answer your original question - no, there isn't a way to see if the user is in an experiment without bucketing them into it that I'm aware of.
You can add targeting conditions, so, for example, if you had a user attribute of hasClickedButtonC, the user would only be bucketed into the experiment (and thus, fire the tracking callback) if that was true. If it wasn't, useIsFeatureEnabled would just return false without firing the tracking callBack.

> One thing that might be helpful, you the Javascript SDK does have a handy setAttributes method - so onClick, you can update the user's attributes. But, you'll want to make sure that the attributes can persist so the next time they land on this page or any others, they have the necessary attributes, otherwise, they might get different variations on subsequent visits.

Reply:
Confirmed with Dave that its ok to trigger/enroll users on history and popular feed as well.